### PR TITLE
Knn in c

### PR DESCRIPTION
**Pull request by aayushbest**
**Hi!**  
Thanks for considering for contributing to this ever-growing list of algorithm and data structure implementations.
Your contribution is valuable. Use the follow template for creating PRs. (Delete this part)
---

**Fixes #{ISSUE_NUMBER}**

By submitting this pull request I confirm I've read and complied with the below declarations.

-   [ ] I have read the [Contribution guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md) 
and I am confident that my PR reflects them.

-   [ ] I have followed the [coding guidelines](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#cs) 
for this project.

-   [ ] My code follows the [skeleton code structure](https://github.com/FOSS-UCSC/FOSSALGO/blob/master/CONTRIBUTING.md#sample).

-   [ ] This pull request has a descriptive title. 
For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.

-   [ ] This pull request shall only be merged once the build passes.

-   [ ] I have mentioned the issue number correctly (with hyper-link) in this pull request description.

### CODE REVIEW

File: algorithms/ar-knn/c/knn.c [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    distance = (float *)malloc(size * sizeof(float));
+    for(int i = 0; i < (int)size; ++i){
+        distance[i] = get_euclidean_distance(training_set[i], q); 
+    }
+}
+// utility function for swapping elements in collection
+void swap(void *x, void *y){
+    void *temp = x;
+    x = y;
+    y = temp;
```
Comment (RavinduSachintha) [inline algorithms/ar-knn/c/knn.c:35]: Codacy found an issue: [Variable 'y' is assigned a value that is never used.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6388877)

Comment (RavinduSachintha) [inline algorithms/ar-knn/c/knn.c:35]: Codacy found an issue: [Assignment of function parameter has no effect outside the function. Did you forget dereferencing it?](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6388877)

File: algorithms/ar-knn/c/knn.c [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+void update_distance(const Point training_set[], const size_t size,const Point q){
+    distance = (float *)malloc(size * sizeof(float));
+    for(int i = 0; i < (int)size; ++i){
+        distance[i] = get_euclidean_distance(training_set[i], q); 
+    }
+}
+// utility function for swapping elements in collection
+void swap(void *x, void *y){
+    void *temp = x;
+    x = y;
```
Comment (RavinduSachintha) [inline algorithms/ar-knn/c/knn.c:34]: Codacy found an issue: [Variable 'x' is assigned a value that is never used.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6388877)

Comment (RavinduSachintha) [inline algorithms/ar-knn/c/knn.c:34]: Codacy found an issue: [Assignment of function parameter has no effect outside the function. Did you forget dereferencing it?](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6388877)

Comment (aayushbest) [reply algorithms/ar-knn/c/knn.c:34]: Yes let me resolve it. Although it was working fine on my gcc compiler

**Comment by aayushbest** [comment]
Hi @Danushka96  There is something wrong with the code checks gcc compiler it is telling undefined sqrt function. Even I have already included math.h library in module which has the function prototype of double sqrt(double X). Please look into the issue.

**Review by Danushka96 — APPROVED** [review]

**Comment by Danushka96** [comment]
Hi @aayushbest 
Thanks for your contribution, Sorry for the compilation issue, And thanks for reporting that!

**Comment by aayushbest** [comment]
The pleasure is all mine. Looking forward to keep my momentum going with the help of your approval and guidance. 😊

**Pull request merged**